localstack-core/localstack/services/lambda_/event_source_mapping/esm_worker_factory.py

```diff
@@ -41,6 +41,7 @@ def __init__(self, esm_config, function_role, enabled):
         self.function_role_arn = function_role
         self.enabled = enabled
 
+    # TODO Should we split up each Poller creation to a separate static function?
```

Comment:
certainly possible, I didn't see a big need for it so far given that factories are typically quite easy to understand